x/staking/README.md

```diff
@@ -270,6 +272,36 @@ A redelegation object is created every time a redelegation occurs. To prevent
 https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc1/proto/cosmos/staking/v1beta1/staking.proto#L263-L308
 ```
 
+## ConsPubkeyRotation
+
+The `ConsPubkey` of a validator will be instantly rotated to the new `ConsPubkey` there will be a tracking for the rotations which is only allows limited no.of rotations should be made in the unbonding period of time
+
+`ConsPubkeyRotation` are indexed in the store as:
+
+ConsPubKeyRotationHistory: `0x70 | validatorAddrLen(1 byte) | validatorAddr | rotated_height -> ProtocalBuffer(ConsPubKeyRotationHistory)`
+ValidatorConsensusKeyRotationRecordQueue: `0x72 | format(time) -> ProtocalBuffer(RotatedValAddrs)`
+
+ValidatorConsensusKeyRotationRecordIndex:`0x73 | valAddressLen(1 byte) | valAddr | format(time) -> ProtocalBuffer([]Byte{})`
+
+`ConsPubKeyRotationHistory` is used for querying the rotations of a validator
+
+`ValidatorConsensusKeyRotationRecordQueue` is to keep track of the rotation across the unbonding period (waiting period in the queue), this be pruned after the unbonding period of waiting time.
+
+`ValidatorConsensusKeyRotationRecordIndex` is to keep track of a validator that how many rotations were made inside unbonding period. This be pruned after the unbonding period of waiting time.
+
+A `ConsPubKeyRotationHistory` object is created every time a consensus pubkey rotation occurs.
+
+To prevent the spam: 
+
+* There will only limited number of rotations can be done within unbonding period of time. 
+* And significant amount of fee will be deducted for rotating the keys.
+
+Formula to calculate the fees to rotate the key is: 
+
+```go
+(max(VotingPowerPercentage 100, 1) * InitialKeyRotationFee) * 2^(number of rotations in ConsPubKeyRotationHistory in recent unbonding period)
```

Comment:
What is `VotingPowerPercentage 100`?

Reply:
my bad, I think this must be VotingPowerRatio.

Reply:
I think it should just be `VotingPowerPercentage`